// @api.video/nodejs-client/lib/api/VideosApi.js
"use strict";
/**
 * @api.video/nodejs-client
 * api.video is an API that encodes on the go to facilitate immediate playback, enhancing viewer streaming experiences across multiple devices and platforms. You can stream live or on-demand online videos within minutes.
 *
 * The version of the OpenAPI document: 1
 *
 *
 * NOTE: This class is auto generated.
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const path_1 = __importDefault(require("path"));
const fs_1 = require("fs");
const url_1 = require("url");
const form_data_1 = __importDefault(require("form-data"));
const ObjectSerializer_1 = __importDefault(require("../ObjectSerializer"));
const stream_1 = require("stream");
const HttpClient_1 = require("../HttpClient");
/**
 * no description
 */
class VideosApi {
    constructor(httpClient) {
        this.httpClient = httpClient;
    }
    /**
     * Creates a video object. More information on video objects can be found [here](https://docs.api.video/reference/videos-1).
     * Create a video object
     * @param videoCreationPayload video to create
     */
    async create(videoCreationPayload) {
        const queryParams = {};
        queryParams.headers = {};
        if (videoCreationPayload === null || videoCreationPayload === undefined) {
            throw new Error('Required parameter videoCreationPayload was null or undefined when calling create.');
        }
        // Path Params
        const localVarPath = '/videos'.substring(1);
        // Body Params
        const contentType = ObjectSerializer_1.default.getPreferredMediaType([
            'application/json',
        ]);
        queryParams.headers['Content-Type'] = contentType;
        queryParams.body = ObjectSerializer_1.default.stringify(ObjectSerializer_1.default.serialize(videoCreationPayload, 'VideoCreationPayload', ''), contentType);
        queryParams.method = 'POST';
        return this.httpClient
            .call(localVarPath, queryParams)
            .then((response) => ObjectSerializer_1.default.deserialize(ObjectSerializer_1.default.parse(response.body, response.headers['content-type']), 'Video', ''));
    }
    /**
     * Upload a video
     * This will create a progressive upload session.
     * @param videoId Enter the videoId you want to use to upload your video.
     */
    createUploadProgressiveSession(videoId) {
        class UploadProgressiveSession {
            constructor(httpClient) {
                this.currentPart = 1;
                this.httpClient = httpClient;
            }
            uploadPart(file, progressListener) {
                return this.upload(file, false, progressListener);
            }
            uploadLastPart(file, progressListener) {
                return this.upload(file, true, progressListener);
            }
            upload(file, isLast, progressListener) {
                const queryParams = {};
                queryParams.headers = {};
                if (videoId === null || videoId === undefined) {
                    throw new Error('Required parameter videoId was null or undefined when calling upload.');
                }
                if (!(0, fs_1.existsSync)(file)) {
                    throw new Error(`${file} must be a readable source file`);
                }
                const length = (0, fs_1.statSync)(file).size;
                if (length <= 0) {
                    throw new Error(`${file} is empty`);
                }
                // Path Params
                const localVarPath = '/videos/{videoId}/source'
                    .substring(1)
                    .replace('{' + 'videoId' + '}', encodeURIComponent(String(videoId)));
                queryParams.method = 'POST';
                const formData = new form_data_1.default();
                const filename = path_1.default.basename(file);
                formData.append(filename, (0, fs_1.createReadStream)(file), filename);
                queryParams.body = formData;
                queryParams.headers['Content-Range'] = `part ${this.currentPart}/${isLast ? this.currentPart : '*'}`;
                if (progressListener) {
                    queryParams.onUploadProgress = (progress) => {
                        progressListener({
                            chunksCount: 1,
                            currentChunk: 0,
                            currentChunkUploadedBytes: progress.loaded,
                            currentChunkTotalBytes: progress.total || 0,
                            uploadedBytes: progress.loaded,
                            totalBytes: progress.total || 0,
                        });
                    };
                }
                const call = this.httpClient.call(localVarPath, queryParams);
                this.currentPart++;
                return call.then((response) => ObjectSerializer_1.default.deserialize(ObjectSerializer_1.default.parse(response.body, response.headers['content-type']), 'Video', ''));
            }
        }
        return new UploadProgressiveSession(this.httpClient);
    }
    /**
     * To upload a video to the videoId you created. You can only upload your video to the videoId once.
  
  
  
  We offer 2 types of upload:
  
  * Regular upload
  
  * Progressive upload
  
  The latter allows you to split a video source into X chunks and send those chunks independently (concurrently or sequentially). The 2 main goals for our users are to
  
    * allow the upload of video sources > 200 MiB (200 MiB = the max. allowed file size for regular upload)
  
    * allow to send a video source "progressively", i.e., before before knowing the total size of the video.
  
    Once all chunks have been sent, they are reaggregated to one source file. The video source is considered as "completely sent" when the "last" chunk is sent (i.e., the chunk that "completes" the upload).
  
  
     * Upload a video
     * @param videoId Enter the videoId you want to use to upload your video.
     * @param file The path to the video you would like to upload. The path must be local. If you want to use a video from an online source, you must use the \\\&quot;/videos\\\&quot; endpoint and add the \\\&quot;source\\\&quot; parameter when you create a new video.
     */
    async upload(videoId, file, progressListener) {
        const queryParams = {};
        queryParams.headers = {};
        if (videoId === null || videoId === undefined) {
            throw new Error('Required parameter videoId was null or undefined when calling upload.');
        }
        if (!(0, fs_1.existsSync)(file)) {
            throw new Error(`${file} must be a readable source file`);
        }
        const length = (0, fs_1.statSync)(file).size;
        if (length <= 0) {
            throw new Error(`${file} is empty`);
        }
        // Path Params
        const localVarPath = '/videos/{videoId}/source'
            .substring(1)
            .replace('{' + 'videoId' + '}', encodeURIComponent(String(videoId)));
        queryParams.method = 'POST';
        const formData = new form_data_1.default();
        const chunkSize = this.httpClient.getChunkSize();
        // Upload in a single request when file is small enough
        if (chunkSize > length) {
            const filename = path_1.default.basename(file);
            formData.append(filename, (0, fs_1.createReadStream)(file), filename);
            queryParams.body = formData;
            if (progressListener) {
                queryParams.onUploadProgress = (progress) => {
                    progressListener({
                        chunksCount: 1,
                        currentChunk: 0,
                        currentChunkUploadedBytes: progress.loaded,
                        currentChunkTotalBytes: progress.total || 0,
                        uploadedBytes: progress.loaded,
                        totalBytes: progress.total || 0,
                    });
                };
            }
            const call = this.httpClient.call(localVarPath, queryParams);
            return call.then((response) => ObjectSerializer_1.default.deserialize(ObjectSerializer_1.default.parse(response.body, response.headers['content-type']), 'Video', ''));
        }
        let uploadChunkSize = chunkSize;
        let lastBody;
        let stream;
        let chunkNumber = 0;
        const partsCount = Math.ceil(length / chunkSize);
        let part = 1;
        for (let offset = 0; offset < length; offset += chunkSize, chunkNumber++) {
            // default the upload size to be as large as possible.
            uploadChunkSize = chunkSize;
            // BUT,if we are on the last chunk to be uploaded, the uploaded chunk must be
            // reduced to match the remaining bytes in the file
            if (offset + uploadChunkSize > length) {
                uploadChunkSize = length - offset;
            }
            const filename = path_1.default.basename(file);
            const chunkFormData = new form_data_1.default();
            stream = (0, fs_1.createReadStream)(file, {
                start: offset,
                end: uploadChunkSize + offset - 1,
            });
            chunkFormData.append(filename, stream, filename);
            queryParams.body = chunkFormData;
            queryParams.headers['Content-Range'] = `part ${part}/${partsCount}`;
            part++;
            if (progressListener) {
                queryParams.onUploadProgress = (progress) => {
                    progressListener({
                        chunksCount: Math.ceil(length / chunkSize),
                        currentChunk: chunkNumber,
                        currentChunkUploadedBytes: progress.loaded,
                        currentChunkTotalBytes: progress.total || 0,
                        uploadedBytes: Math.min(length, chunkNumber * chunkSize + progress.loaded),
                        totalBytes: length,
                    });
                };
            }
            const call = this.httpClient.call(localVarPath, queryParams);
            lastBody = await call.then((response) => ObjectSerializer_1.default.deserialize(ObjectSerializer_1.default.parse(response.body, response.headers['content-type']), 'Video', ''));
            stream.close();
        }
        return Promise.resolve(lastBody);
    }
    /**
     * Upload with an delegated upload token
     * This will create a progressive upload session.
     * @param token The unique identifier for the token you want to use to upload a video.
     */
    createUploadWithUploadTokenProgressiveSession(token) {
        class UploadWithUploadTokenProgressiveSession {
            constructor(httpClient) {
                this.currentPart = 1;
                this.httpClient = httpClient;
            }
            uploadPart(file, progressListener) {
                return this.upload(file, false, progressListener);
            }
            uploadLastPart(file, progressListener) {
                return this.upload(file, true, progressListener);
            }
            upload(file, isLast, progressListener) {
                const queryParams = {};
                queryParams.headers = {};
                if (token === null || token === undefined) {
                    throw new Error('Required parameter token was null or undefined when calling uploadWithUploadToken.');
                }
                if (!(0, fs_1.existsSync)(file)) {
                    throw new Error(`${file} must be a readable source file`);
                }
                const length = (0, fs_1.statSync)(file).size;
                if (length <= 0) {
                    throw new Error(`${file} is empty`);
                }
                // Path Params
                const localVarPath = '/upload'.substring(1);
                // Query Params
                const urlSearchParams = new url_1.URLSearchParams();
                if (token !== undefined) {
                    urlSearchParams.append('token', ObjectSerializer_1.default.serialize(token, 'string', ''));
                }
                queryParams.searchParams = urlSearchParams;
                queryParams.method = 'POST';
                const formData = new form_data_1.default();
                const filename = path_1.default.basename(file);
                formData.append(filename, (0, fs_1.createReadStream)(file), filename);
                if (this.videoId) {
                    formData.append('videoId', this.videoId);
                }
                queryParams.body = formData;
                queryParams.headers['Content-Range'] = `part ${this.currentPart}/${isLast ? this.currentPart : '*'}`;
                if (progressListener) {
                    queryParams.onUploadProgress = (progress) => {
                        progressListener({
                            chunksCount: 1,
                            currentChunk: 0,
                            currentChunkUploadedBytes: progress.loaded,
                            currentChunkTotalBytes: progress.total || 0,
                            uploadedBytes: progress.loaded,
                            totalBytes: progress.total || 0,
                        });
                    };
                }
                const call = this.httpClient.call(localVarPath, queryParams);
                this.currentPart++;
                return call
                    .then((response) => ObjectSerializer_1.default.deserialize(ObjectSerializer_1.default.parse(response.body, response.headers['content-type']), 'Video', ''))
                    .then((res) => {
                    this.videoId = res.videoId;
                    return res;
                });
            }
        }
        return new UploadWithUploadTokenProgressiveSession(this.httpClient);
    }
    /**
     * This method allows you to send a video using an upload token. Upload tokens are especially useful when the upload is done from the client side. If you want to upload a video from your server-side application, you'd better use the [standard upload method](#upload).
     * Upload with an delegated upload token
     * @param token The unique identifier for the token you want to use to upload a video.
     * @param file The path to the video you want to upload.
     */
    async uploadWithUploadToken(token, file, progressListener) {
        const queryParams = {};
        queryParams.headers = {};
        if (token === null || token === undefined) {
            throw new Error('Required parameter token was null or undefined when calling uploadWithUploadToken.');
        }
        if (!(0, fs_1.existsSync)(file)) {
            throw new Error(`${file} must be a readable source file`);
        }
        const length = (0, fs_1.statSync)(file).size;
        if (length <= 0) {
            throw new Error(`${file} is empty`);
        }
        // Path Params
        const localVarPath = '/upload'.substring(1);
        // Query Params
        const urlSearchParams = new url_1.URLSearchParams();
        if (token !== undefined) {
            urlSearchParams.append('token', ObjectSerializer_1.default.serialize(token, 'string', ''));
        }
        queryParams.searchParams = urlSearchParams;
        queryParams.method = 'POST';
        const formData = new form_data_1.default();
        const chunkSize = this.httpClient.getChunkSize();
        // Upload in a single request when file is small enough
        if (chunkSize > length) {
            const filename = path_1.default.basename(file);
            formData.append(filename, (0, fs_1.createReadStream)(file), filename);
            queryParams.body = formData;
            if (progressListener) {
                queryParams.onUploadProgress = (progress) => {
                    progressListener({
                        chunksCount: 1,
                        currentChunk: 0,
                        currentChunkUploadedBytes: progress.loaded,
                        currentChunkTotalBytes: progress.total || 0,
                        uploadedBytes: progress.loaded,
                        totalBytes: progress.total || 0,
                    });
                };
            }
            const call = this.httpClient.call(localVarPath, queryParams);
            return call.then((response) => ObjectSerializer_1.default.deserialize(ObjectSerializer_1.default.parse(response.body, response.headers['content-type']), 'Video', ''));
        }
        let uploadChunkSize = chunkSize;
        let lastBody;
        let stream;
        let chunkNumber = 0;
        const partsCount = Math.ceil(length / chunkSize);
        let part = 1;
        for (let offset = 0; offset < length; offset += chunkSize, chunkNumber++) {
            // default the upload size to be as large as possible.
            uploadChunkSize = chunkSize;
            // BUT,if we are on the last chunk to be uploaded, the uploaded chunk must be
            // reduced to match the remaining bytes in the file
            if (offset + uploadChunkSize > length) {
                uploadChunkSize = length - offset;
            }
            const filename = path_1.default.basename(file);
            const chunkFormData = new form_data_1.default();
            stream = (0, fs_1.createReadStream)(file, {
                start: offset,
                end: uploadChunkSize + offset - 1,
            });
            chunkFormData.append(filename, stream, filename);
            queryParams.body = chunkFormData;
            queryParams.headers['Content-Range'] = `part ${part}/${partsCount}`;
            part++;
            if (progressListener) {
                queryParams.onUploadProgress = (progress) => {
                    progressListener({
                        chunksCount: Math.ceil(length / chunkSize),
                        currentChunk: chunkNumber,
                        currentChunkUploadedBytes: progress.loaded,
                        currentChunkTotalBytes: progress.total || 0,
                        uploadedBytes: Math.min(length, chunkNumber * chunkSize + progress.loaded),
                        totalBytes: length,
                    });
                };
            }
            const call = this.httpClient.call(localVarPath, queryParams);
            lastBody = await call.then((response) => ObjectSerializer_1.default.deserialize(ObjectSerializer_1.default.parse(response.body, response.headers['content-type']), 'Video', ''));
            stream.close();
        }
        return Promise.resolve(lastBody);
    }
    /**
     * This call provides the same information provided on video creation. For private videos, it will generate a unique token url. Use this to retrieve any details you need about a video, or set up a private viewing URL.
     * Retrieve a video object
     * @param videoId The unique identifier for the video you want details about.
     */
    async get(videoId) {
        const queryParams = {};
        queryParams.headers = {};
        if (videoId === null || videoId === undefined) {
            throw new Error('Required parameter videoId was null or undefined when calling get.');
        }
        // Path Params
        const localVarPath = '/videos/{videoId}'
            .substring(1)
            .replace('{' + 'videoId' + '}', encodeURIComponent(String(videoId)));
        queryParams.method = 'GET';
        return this.httpClient
            .call(localVarPath, queryParams)
            .then((response) => ObjectSerializer_1.default.deserialize(ObjectSerializer_1.default.parse(response.body, response.headers['content-type']), 'Video', ''));
    }
    /**
     * Updates the parameters associated with a video ID. The video object you are updating is determined by the video ID you provide.
  
  
  
  NOTE: If you are updating an array, you must provide the entire array as what you provide here overwrites what is in the system rather than appending to it.
  
  
     * Update a video object
     * @param videoId The video ID for the video you want to update.
     * @param videoUpdatePayload
     */
    async update(videoId, videoUpdatePayload = {}) {
        const queryParams = {};
        queryParams.headers = {};
        if (videoId === null || videoId === undefined) {
            throw new Error('Required parameter videoId was null or undefined when calling update.');
        }
        if (videoUpdatePayload === null || videoUpdatePayload === undefined) {
            throw new Error('Required parameter videoUpdatePayload was null or undefined when calling update.');
        }
        // Path Params
        const localVarPath = '/videos/{videoId}'
            .substring(1)
            .replace('{' + 'videoId' + '}', encodeURIComponent(String(videoId)));
        // Body Params
        const contentType = ObjectSerializer_1.default.getPreferredMediaType([
            'application/json',
        ]);
        queryParams.headers['Content-Type'] = contentType;
        queryParams.body = ObjectSerializer_1.default.stringify(ObjectSerializer_1.default.serialize(videoUpdatePayload, 'VideoUpdatePayload', ''), contentType);
        queryParams.method = 'PATCH';
        return this.httpClient
            .call(localVarPath, queryParams)
            .then((response) => ObjectSerializer_1.default.deserialize(ObjectSerializer_1.default.parse(response.body, response.headers['content-type']), 'Video', ''));
    }
    /**
     * If you do not need a video any longer, you can send a request to delete it. All you need is the videoId.
     * Delete a video object
     * @param videoId The video ID for the video you want to delete.
     */
    async delete(videoId) {
        const queryParams = {};
        queryParams.headers = {};
        if (videoId === null || videoId === undefined) {
            throw new Error('Required parameter videoId was null or undefined when calling delete.');
        }
        // Path Params
        const localVarPath = '/videos/{videoId}'
            .substring(1)
            .replace('{' + 'videoId' + '}', encodeURIComponent(String(videoId)));
        queryParams.method = 'DELETE';
        return this.httpClient
            .call(localVarPath, queryParams)
            .then((response) => ObjectSerializer_1.default.deserialize(ObjectSerializer_1.default.parse(response.body, response.headers['content-type']), 'void', ''));
    }
    /**
     * This method returns a list of your videos (with all their details). With no parameters added, the API returns the first page of all videos. You can filter videos using the parameters described below.
     * List all video objects
     * @param {Object} searchParams
     * @param { string } searchParams.title The title of a specific video you want to find. The search will match exactly to what term you provide and return any videos that contain the same term as part of their titles.
     * @param { Array&lt;string&gt; } searchParams.tags A tag is a category you create and apply to videos. You can search for videos with particular tags by listing one or more here. Only videos that have all the tags you list will be returned.
     * @param { { [key: string]: string; } } searchParams.metadata Videos can be tagged with metadata tags in key:value pairs. You can search for videos with specific key value pairs using this parameter. [Dynamic Metadata](https://api.video/blog/endpoints/dynamic-metadata) allows you to define a key that allows any value pair.
     * @param { string } searchParams.description Retrieve video objects by &#x60;description&#x60;.
     * @param { string } searchParams.liveStreamId Retrieve video objects that were recorded from a live stream by &#x60;liveStreamId&#x60;.
     * @param { &#39;title&#39; | &#39;createdAt&#39; | &#39;publishedAt&#39; | &#39;updatedAt&#39; } searchParams.sortBy Use this parameter to sort videos by the their created time, published time, updated time, or by title.
     * @param { &#39;asc&#39; | &#39;desc&#39; } searchParams.sortOrder Use this parameter to sort results. &#x60;asc&#x60; is ascending and sorts from A to Z. &#x60;desc&#x60; is descending and sorts from Z to A.
     * @param { number } searchParams.currentPage Choose the number of search results to return per page. Minimum value: 1
     * @param { number } searchParams.pageSize Results per page. Allowed values 1-100, default is 25.
     */
    async list({ title, tags, metadata, description, liveStreamId, sortBy, sortOrder, currentPage, pageSize, } = {}) {
        const queryParams = {};
        queryParams.headers = {};
        // Path Params
        const localVarPath = '/videos'.substring(1);
        // Query Params
        const urlSearchParams = new url_1.URLSearchParams();
        if (title !== undefined) {
            urlSearchParams.append('title', ObjectSerializer_1.default.serialize(title, 'string', ''));
        }
        if (tags !== undefined) {
            const tagsSerialized = ObjectSerializer_1.default.serialize(tags, 'Array<string>', '');
            tagsSerialized.forEach((v) => urlSearchParams.append('tags[]', v));
        }
        if (metadata !== undefined) {
            if (typeof metadata !== 'object') {
                throw new Error(`${metadata} is not an object`);
            }
            Object.keys(metadata).forEach((k) => urlSearchParams.append('metadata[' + k + ']', ObjectSerializer_1.default.serialize(metadata[k], 'string', '')));
        }
        if (description !== undefined) {
            urlSearchParams.append('description', ObjectSerializer_1.default.serialize(description, 'string', ''));
        }
        if (liveStreamId !== undefined) {
            urlSearchParams.append('liveStreamId', ObjectSerializer_1.default.serialize(liveStreamId, 'string', ''));
        }
        if (sortBy !== undefined) {
            urlSearchParams.append('sortBy', ObjectSerializer_1.default.serialize(sortBy, "'title' | 'createdAt' | 'publishedAt' | 'updatedAt'", ''));
        }
        if (sortOrder !== undefined) {
            urlSearchParams.append('sortOrder', ObjectSerializer_1.default.serialize(sortOrder, "'asc' | 'desc'", ''));
        }
        if (currentPage !== undefined) {
            urlSearchParams.append('currentPage', ObjectSerializer_1.default.serialize(currentPage, 'number', ''));
        }
        if (pageSize !== undefined) {
            urlSearchParams.append('pageSize', ObjectSerializer_1.default.serialize(pageSize, 'number', ''));
        }
        queryParams.searchParams = urlSearchParams;
        queryParams.method = 'GET';
        return this.httpClient
            .call(localVarPath, queryParams)
            .then((response) => ObjectSerializer_1.default.deserialize(ObjectSerializer_1.default.parse(response.body, response.headers['content-type']), 'VideosListResponse', ''));
    }
    /**
     * The thumbnail is the poster that appears in the player window before video playback begins.
  
  
  
  This endpoint allows you to upload an image for the thumbnail.
  
  
  
  To select a still frame from the video using a time stamp, use the [dedicated method](#pickThumbnail) to pick a time in the video.
  
  
  
  Note: There may be a short delay before the new thumbnail is delivered to our CDN.
     * Upload a thumbnail
     * @param videoId Unique identifier of the chosen video
     * @param file The image to be added as a thumbnail. The mime type should be image/jpeg, image/png or image/webp. The max allowed size is 8 MiB.
     */
    async uploadThumbnail(videoId, file) {
        const queryParams = {};
        queryParams.headers = {};
        if (videoId === null || videoId === undefined) {
            throw new Error('Required parameter videoId was null or undefined when calling uploadThumbnail.');
        }
        let fileName = 'file';
        let fileBuffer = file;
        if (typeof file === 'string') {
            fileName = path_1.default.basename(file);
            fileBuffer = (0, fs_1.createReadStream)(file);
        }
        if (file instanceof stream_1.Readable) {
            fileBuffer = await (0, HttpClient_1.readableToBuffer)(file);
        }
        // Path Params
        const localVarPath = '/videos/{videoId}/thumbnail'
            .substring(1)
            .replace('{' + 'videoId' + '}', encodeURIComponent(String(videoId)));
        queryParams.method = 'POST';
        const formData = new form_data_1.default();
        formData.append(fileName, fileBuffer, fileName);
        queryParams.body = formData;
        return this.httpClient
            .call(localVarPath, queryParams)
            .then((response) => ObjectSerializer_1.default.deserialize(ObjectSerializer_1.default.parse(response.body, response.headers['content-type']), 'Video', ''));
    }
    /**
     * Pick a thumbnail from the given time code.
  
  
  
  If you'd like to upload an image for your thumbnail, use the dedicated [method](#uploadThumbnail).
  
  
  
  There may be a short delay for the thumbnail to update.
  
  
     * Set a thumbnail
     * @param videoId Unique identifier of the video you want to add a thumbnail to, where you use a section of your video as the thumbnail.
     * @param videoThumbnailPickPayload
     */
    async pickThumbnail(videoId, videoThumbnailPickPayload) {
        const queryParams = {};
        queryParams.headers = {};
        if (videoId === null || videoId === undefined) {
            throw new Error('Required parameter videoId was null or undefined when calling pickThumbnail.');
        }
        if (videoThumbnailPickPayload === null ||
            videoThumbnailPickPayload === undefined) {
            throw new Error('Required parameter videoThumbnailPickPayload was null or undefined when calling pickThumbnail.');
        }
        // Path Params
        const localVarPath = '/videos/{videoId}/thumbnail'
            .substring(1)
            .replace('{' + 'videoId' + '}', encodeURIComponent(String(videoId)));
        // Body Params
        const contentType = ObjectSerializer_1.default.getPreferredMediaType([
            'application/json',
        ]);
        queryParams.headers['Content-Type'] = contentType;
        queryParams.body = ObjectSerializer_1.default.stringify(ObjectSerializer_1.default.serialize(videoThumbnailPickPayload, 'VideoThumbnailPickPayload', ''), contentType);
        queryParams.method = 'PATCH';
        return this.httpClient
            .call(localVarPath, queryParams)
            .then((response) => ObjectSerializer_1.default.deserialize(ObjectSerializer_1.default.parse(response.body, response.headers['content-type']), 'Video', ''));
    }
    /**
     * This method provides upload status & encoding status to determine when the video is uploaded or ready to playback. Once encoding is completed, the response also lists the available stream qualities.
     * Retrieve video status and details
     * @param videoId The unique identifier for the video you want the status for.
     */
    async getStatus(videoId) {
        const queryParams = {};
        queryParams.headers = {};
        if (videoId === null || videoId === undefined) {
            throw new Error('Required parameter videoId was null or undefined when calling getStatus.');
        }
        // Path Params
        const localVarPath = '/videos/{videoId}/status'
            .substring(1)
            .replace('{' + 'videoId' + '}', encodeURIComponent(String(videoId)));
        queryParams.method = 'GET';
        return this.httpClient
            .call(localVarPath, queryParams)
            .then((response) => ObjectSerializer_1.default.deserialize(ObjectSerializer_1.default.parse(response.body, response.headers['content-type']), 'VideoStatus', ''));
    }
}
exports.default = VideosApi;
