// @api.video/nodejs-client/lib/model/RestreamsRequestObject.js
"use strict";
/**
 * @api.video/nodejs-client
 * api.video is an API that encodes on the go to facilitate immediate playback, enhancing viewer streaming experiences across multiple devices and platforms. You can stream live or on-demand online videos within minutes.
 *
 * The version of the OpenAPI document: 1
 *
 *
 * NOTE: This class is auto generated.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * Adding restream destinations is optional. However, if you set a restream destination, you must provide all attributes for each destination.
 */
class RestreamsRequestObject {
    static getAttributeTypeMap() {
        return RestreamsRequestObject.attributeTypeMap;
    }
}
RestreamsRequestObject.discriminator = undefined;
RestreamsRequestObject.attributeTypeMap = [
    {
        name: 'name',
        baseName: 'name',
        type: 'string',
        format: '',
    },
    {
        name: 'serverUrl',
        baseName: 'serverUrl',
        type: 'string',
        format: '',
    },
    {
        name: 'streamKey',
        baseName: 'streamKey',
        type: 'string',
        format: '',
    },
];
exports.default = RestreamsRequestObject;
